lazy_tensor_core/lazy_tensors/core/platform/hash.cc

```diff
@@ -0,0 +1,72 @@
+#include "lazy_tensors/core/platform/hash.h"
+
+#include <cstring>
+
+#include "lazy_tensors/core/platform/macros.h"
+
+namespace lazy_tensors {
+namespace {
+
+// 0xff is in case char is signed.
+inline uint64 ByteAs64(char c) { return static_cast<uint64>(c) & 0xff; }
+
+uint64 DecodeFixed64(const char* ptr) {
+  // Load the raw bytes
+  uint64 result;
+  memcpy(&result, ptr, sizeof(result));  // gcc optimizes this to a plain load
+  return result;
+}
+
+}  // namespace
+
+uint64 Hash64(const char* data, size_t n, uint64 seed) {
```

Comment:
This is taken from TF and makes it possible to compare the XLA plugin with PT/XLA, which was useful (and might still be) for debugging. It doesn't say what algorithm this is and git blame points to the initial commit 6 years ago.

We can either use MurmurHash since it doesn't need to be cryptographic or credit TF authors.

Reply:
ok, pointer to TF commit would be fine i think.  I support the idea of not changing behavior now and maintaining debug-ability.  If its a direct copy paste of something working 6 years and covered by backend tests then I'm ok with it.